extensions/amp-cld-img/0.1/utils.js

```diff
@@ -0,0 +1,376 @@
+        'resourceType', 'image'));
+    let version = /** @type {?string} */(optionConsume(options, 'version',
+        null));
+    let transformation = generateTransformationString(options);
+    const format = /** @type {?string} */(optionConsume(options, 'format',
+        null));
+    const cloudName = /** @type {string} */(optionConsume(options, 'cloudName',
+        null));
+    const privateCdn = /** @type {?boolean} */(optionConsume(options,
+        'privateCdn', null));
+    const secureDistribution = /** @type {?string} */(optionConsume(options,
+        'secureDistribution', null));
+    const secure = /** @type {?boolean} */(optionConsume(options, 'secure',
+        true));
+    const cdnSubdomain = /** @type {?string} */(optionConsume(options,
+        'cdnSubdomain', null));
+    const secureCdnSubdomain = /** @type {?boolean} */(optionConsume(options,
+        'secureCdnSubdomain', null));
+    const cname = /** @type {?string} */(optionConsume(options, 'cname', null));
+    const shorten = /** @type {?boolean} */(optionConsume(options, 'shorten',
+        null));
+    const urlSuffix = /** @type {?string} */(optionConsume(options, 'urlSuffix',
+        null));
+    const useRootPath = /** @type {?boolean} */(optionConsume(options,
+        'useRootPath', null));
+
+    const preloaded = /^(image|raw)\/([a-z0-9_]+)\/v(\d+)\/([^#]+)$/.exec(
+        publicId);
+    if (preloaded) {
+      resourceType = preloaded[1];
+      type = preloaded[2];
+      version = preloaded[3];
+      publicId = preloaded[4];
+    }
+    const originalSource = publicId;
+    if (publicId == null) {
+      return originalSource;
+    }
+
+    if (type === null && publicId.match(/^https?:\//i)) {
+      return originalSource;
+    }
+    const finalizedResourceType = finalizeResourceType(resourceType, type,
+        urlSuffix, useRootPath, shorten);
+
+    resourceType = finalizedResourceType[0];
+    type = finalizedResourceType[1];
+
+    const finalizedSource = finalizeSource(publicId, format, urlSuffix);
+    publicId = finalizedSource[0];
+    const sourceToSign = finalizedSource[1];
+
+    if (sourceToSign.indexOf('/') > 0 && !sourceToSign.match(/^v[0-9]+/) &&
+      !sourceToSign.match(/^https?:\//)) {
+      if (version == null) {
+        version = 1;
+      }
+    }
+    if (version != null) {
+      version = `v${version}`;
+    }
+    transformation = transformation.replace(/([^:])\/\//g, '$1/');
+
+    const prefix = unsignedUrlPrefix(publicId, cloudName, privateCdn,
+        cdnSubdomain, secureCdnSubdomain, cname, secure, secureDistribution);
+
+    url = [prefix, resourceType, type, transformation,
+      version, publicId].filter(function(part) {
+      return (part != null) && part !== '';
+    }).join('/');
+  }
+
+  return url;
+}
+
+/**
+ * Converts the string to boolean
+ * @param {string} value to convert to boolean
+ */
+export function getAsBoolean(value) {
+  return value === '1' ||
+    value === 'TRUE' ||
```

Comment:
This doesn't seem consistent with other extensions. Is there a strong need to support different case combinations here? Otherwise, we should support just lowercase.

Also, validation rules should enforce that the things are either "true" or "false" for those options that need a boolean.

I'm also not sure that we should support "1" as meaning boolean true.